editoast/src/views/mod.rs

```diff
@@ -112,6 +116,56 @@ editoast_common::schemas! {
     work_schedules::schemas(),
 }
 
+pub type Roles = editoast_authz::roles::RoleConfig<BuiltinRole>;
+
+// This function will become a middleware once we switch to axum
+async fn make_authorizer<'config>(
+    headers: &axum::http::HeaderMap,
+    roles: &'config Roles,
+    db_pool: Arc<DbConnectionPoolV2>,
+) -> Result<Authorizer<'config, PgAuthDriver<BuiltinRole>>, AuthorizationError> {
+    if roles.is_superuser() {
+        return Ok(Authorizer::new_superuser(
+            roles,
+            PgAuthDriver::<BuiltinRole>::new(db_pool.clone()),
+        ));
+    }
+    let Some(header) = headers.get("x-remote-user") else {
+        return Err(AuthorizationError::Unauthenticated);
+    };
+    let (identity, name) = header
+        .to_str()
+        .expect("unexpected non-ascii characters in x-remote-user")
+        .split_once('/') // FIXME: the gateway should inject two headers instead
```

Comment:
Probably for a follow-up commit or PR, but leaving a comment to not forget about it.